### DISCUSSION TITLE
Any way to set the highest header level in output?

### DISCUSSION THREAD

**Discussion by colleen05** [wooorm/markdown-rs#106] [Q&A] [answered]
I'm thinking of using this for my static site generator, where articles are written in Markdown and converted to static HTML.

Because websites should really only have one `<h1>` header, but titles of Markdown documents may have their header start with a single `#`, I would like to be able to set the highest header "level", such as `#` being `<h2>`, `##` being `<h3>`, etc...

Example:
```md
# Hello, world!
## My section
Lorem ipsum dolor sit amet.
```
would turn into
```html
<h2>Hello, world!</h2>
<h3>My section</h3>
<p>Lorem ipsum dolor sit amet.</p>
```

Is this at all possible? Or is there already an easy way to parse and replace HTML tags with another Rust library?

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @colleen05! 👋 
`markdown-rs` has sibling projects in the JavaScript ecosystem `remark` and `rehype` https://unifiedjs.com/learn/
Those can shift heading levels using https://github.com/rehypejs/rehype-shift-heading

Something similar would be possible in `markdown-rs` after plugins (https://github.com/wooorm/markdown-rs/issues/32) are supported. Insights and code moving plugin support forward are welcome!

**Reply by colleen05** [reply]
Thanks for letting me know about rehype!
Not quite what I'm looking for (as I would prefer to have an entirely Rust codebase and am not big on JavaScript in general), although I may be able to port the heading shifting behaviour to Rust as a standalone library.

I still got my answer, though (i.e., it's not possible just yet with `markdown-rs`, but could be through plugins), so thanks!
